src/Mapbender/WmsBundle/Component/WmsInstanceEntityHandler.php

```diff
@@ -99,30 +109,38 @@ public function setParameters(array $configuration = array())
                 $legendUrl->setHeight(null);
                 $onlineResource = new OnlineResource();
                 $onlineResource->setFormat(null);
-                $onlineResource->setHref($layerDef["legendurl"]);
+                $onlineResource->setHref($configuration["legendurl"]);
                 $legendUrl->setOnlineResource($onlineResource);
                 $style->setLegendUrl($legendUrl);
-                $layersource->addStyle($style);
+                $childLayerSource->addStyle($style);
             }
-            $layersourceroot->addSublayer($layersource);
-            $source->addLayer($layersource);
-            $layerInst       = new WmsInstanceLayer();
-            $layerInst->setTitle($layerDef["title"])
-                ->setId($this->entity->getId() . '_' . $num)
-                ->setMinScale(!isset($layerDef["minScale"]) ? null : $layerDef["minScale"])
-                ->setMaxScale(!isset($layerDef["maxScale"]) ? null : $layerDef["maxScale"])
-                ->setSelected(!isset($layerDef["visible"]) ? false : $layerDef["visible"])
-                ->setInfo(!isset($layerDef["queryable"]) ? false : $layerDef["queryable"])
-                ->setParent($rootInstLayer)
-                ->setSourceItem($layersource)
+            $layersource->addSublayer($childLayerSource);
+            $source->addLayer($childLayerSource);
+            $childLayerInstance       = new WmsInstanceLayer();
+            $childLayerInstance->setTitle($configuration["title"])
+                ->setId($configuration['title'] . '_' . $num)
```

Comment:
Not safe to use title, see above. Since it has always been equal to the layer source's id, it should be explicitly
```php
    ->setId($childLayerSource->getId())
```
... to avoid any further confusion